test/src/sdk-harness/test_artifacts/storage_vec/svec_u8/src/main.sw

```diff
@@ -0,0 +1,73 @@
+contract;
+
+use std::option::*;
+use std::result::*;
+use std::storage::{StorageVec, StorageVecError};
+
+abi MyContract {
+    #[storage(read, write)]
+    fn vec_u8_push(value: u8);
+    #[storage(read)]
+    fn vec_u8_get(index: u64) -> u8;
+    #[storage(read, write)]
+    fn vec_u8_pop() -> u8;
+    #[storage(read, write)]
+    fn vec_u8_remove(index: u64) -> u8;
+    #[storage(read, write)]
+    fn vec_u8_swap_remove(index: u64) -> u8;
+    #[storage(read, write)]
+    fn vec_u8_insert(index: u64, value: u8);
+    #[storage(read)]
+    fn vec_u8_len() -> u64;
+    #[storage(read)]
+    fn vec_u8_is_empty() -> bool;
+    #[storage(write)]
+    fn vec_u8_clear();
+}
+
+storage {
+    vec_u8: StorageVec<u8>,
+}
+
+impl MyContract for Contract {
+    #[storage(read, write)]
+    fn vec_u8_push(value: u8) {
+        storage.vec_u8.push(value);
+    }
+    #[storage(read)]
+    fn vec_u8_get(index: u64) -> u8 {
+        let item: Option<u8> = storage.vec_u8.get(index);
+        item.unwrap()
+    }
+    #[storage(read, write)]
+    fn vec_u8_pop() -> u8 {
+        let res: Option<u8> = storage.vec_u8.pop();
+        res.unwrap()
```

Comment:
```suggestion
    fn vec_u8_pop() -> Option<u8> {
        storage.vec_u8.pop()
```

I think it will panic if you unwrap on a None so you want to return the option and then test that popping with 0 len returns None

Reply:
Thats what i was doing before, but since the rust SDK breaks if you have abi functions that return Results or Options, i switched it to just unwrapping inside the contract

Reply:
v0.16 is out so perhaps it is fixed? I haven't checked. We need to be able to test for None anyway so there's no point in working around this

Reply:
It's still an open issue: https://github.com/FuelLabs/fuels-rs/issues/415

Reply:
So what should i do then, cant return an option so either i work around or im blocked

Reply:
One of the tests needs to be for None so you could either add something to the standard library that is untested or you could wait for the feature to be implemented.
Note that at least @bitzoic would like this feature to exist.

I would say wait and do it properly however it may be beneficial to unblock others and create an issue to track this.
Up to you. @adlerjohn @otrho @mohammadfawaz thoughts?

Reply:
How about using something like https://github.com/FuelLabs/sway/blob/f9fb2a8b7c21655899ae9122a645036527adef41/test/src/sdk-harness/test_projects/exponentiation/mod.rs#L8 to test an expected panic?

Reply:
what would be expected in this case? `Revert(0)` ?

Reply:
Not sure 😂. Try running the test without the `#[should_panic...]` and see if the panic printed out is reasonable. If so, use that as the expected panic reason.